client/login/magic-login/index.jsx

```diff
@@ -132,8 +133,20 @@ class MagicLogin extends Component {
 
 	renderLinks() {
 		const { isJetpackLogin, locale, showCheckYourEmail, translate } = this.props;
-
-		if ( showCheckYourEmail || this.props.query?.client_id ) {
+		const appPromo = (
+			<AppPromo
```

Comment:
This will give you pains if you want to debug later as @sgomes noted here p7H4VZ-4t0-p2.

It's better to just inline the component below. It's the same performance impact because the second instance won't run.

Reply:
Thanks, great suggestion!